### PR TITLE
refactor: adapt browser & node environments

### PR DESCRIPTION
**Pull request by ShookLyngs**
### Description
To publish light-godwoken as an SDK, we need to refactor some parts of it.
In previous version of light-godwoken, we have some codes that can only run on specific envirionment like browser or node:
- Some variables use `process.env.xxx` as their default value, and this is a node specific method
- Some codes use `window.xx` as the value, which is only exist in a browser environment
- The configs of light-godwoken are basically predefined and difficult to customize
- Some features should belong to godwoken-bridge instead of light-godwoken
- Testnet_v0 is deprecated and our test cases are still using the RPC of testnet_v0
- Other issues that rely on the above points

So in this PR, we will basically resolve the above issues by refactoring/fixing the project.

### Light-Godwoken changes
- Organize codes and move them into separate folders
- Replace `window.ethereum` with `EthereumProvider` and refactor relevant codes
- Add `network` param to LightGodwokenProvider and refactor config files for it
- Add `lumosConfig` in LightGodwokenConfig to setup lumos
- Add `getDepositHistories` function to request gwscan api
- Hide testnet_v0 testcases

### Godwoken-Bridge changes
- Move env variables from light-godwoken to godwoken-bridge
- Move `sudtFaucet` from light-godwoken to godwoken-bridge
- Refactor deposit completed list using gwscan's `deposit_histories` api
- Refactor deposit completed list with ahooks (useInfiniteScroll) for pagination

### CODE REVIEW

File: packages/light-godwoken/src/config/config.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    !config.layer2Config ||
+    !config.layer2Config.SCRIPTS ||
+    !config.layer2Config.ROLLUP_CONFIG ||
+    !config.layer2Config.GW_POLYJUICE_RPC_URL ||
+    !config.layer1Config ||
+    !config.layer1Config.SCRIPTS ||
+    !config.layer1Config.CKB_INDEXER_URL ||
+    !config.layer1Config.CKB_RPC_URL
+  ) {
+    throw new LightGodwokenConfigNotValidError(JSON.stringify(config), "config not valid.");
```
Comment (classicalliu) [inline packages/light-godwoken/src/config/config.ts:47]:
```suggestion
    throw new LightGodwokenConfigNotValidError(JSON.stringify(config), "invalid config.");
```

**Review by classicalliu — APPROVED** [review]

File: packages/light-godwoken/src/LightGodwokenV0.ts [hunk 2]
```diff
@@ -303,19 +309,23 @@ export default class DefaultLightGodwokenV0 extends DefaultLightGodwoken impleme
   }
 
   getV1DepositLock(): Script {
-    const lightGodwokenV1 = new DefaultLightGodwokenV1(
-      new DefaultLightGodwokenProvider(this.provider.l2Address, this.provider.ethereum, "v1"),
+    const provider = new DefaultLightGodwokenProvider(
+      this.provider.l2Address,
+      this.provider.ethereum,
+      this.getNetwork(),
+      this.getVersion(),
```
Comment (Flouse) [inline packages/light-godwoken/src/LightGodwokenV0.ts:316]:
```suggestion
      "v1",
```

Comment (Flouse) [reply packages/light-godwoken/src/LightGodwokenV0.ts:316]: I think we should create a v1 provider here.

**Review by Flouse — CHANGES_REQUESTED** [review]

**Review by MrPiNkY2022 — COMMENTED** [review]
This is what I was given so can y'all please help me with the project

**Comment by Flouse** [comment]
> This is what I was given so can y'all please help me with the project

Could you [create an issue](https://github.com/nervosnetwork/light-godwoken/issues/new) to describe what you need?

**Review by Flouse — APPROVED** [review]
LGTM

**Comment by ShookLyngs** [comment]
To prevent potential issues caused by mixing v0/v1 versions of LightGodwoken, I've added some code to make sure lumos helper functions always use the correct LumosConfig.

**Review by RetricSu — APPROVED** [review]

File: packages/light-godwoken/src/__tests__/client.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { providers } from "ethers";
+
+export const lightGodwokenVersionMap = {
+  v0: LightGodwokenV0,
+  v1: LightGodwokenV1,
+};
+
+type LightGodwokenVersionMap = typeof lightGodwokenVersionMap;
+
+export function createLightGodwoken<T extends keyof typeof lightGodwokenVersionMap>(
```
[the remark this answers is not in the document]
Comment (Flouse) [reply packages/light-godwoken/src/__tests__/client.ts:16]: cc @ShookLyngs

[the remark this answers is not in the document]
Comment (ShookLyngs) [reply packages/light-godwoken/src/__tests__/client.ts:16]: > does this works?

Oh yes it works, this is simpler. I can create PR to fix it afterwards.

**Pull request merged**